core/src/main/scala/dagr/core/cmdline/DagrCoreMain.scala

```diff
@@ -239,8 +199,94 @@ class DagrCoreMain(
     interactiveReporter.foreach(_.shutdown())
 
     // return an exit code based on the number of non-completed tasks
-    taskMan.taskToInfoBiMapFor.count { case (task, info) =>
+    taskMan.taskToInfoBiMapFor.count { case (_, info) =>
       TaskStatus.isTaskNotDone(info.status, failedIsDone=false)
     }
   }
+
+}
+
+class DagrCoreMain extends LazyLogging {
+  protected def name: String = "dagr"
+
+  /** A main method that invokes System.exit with the exit code. */
+  def makeItSoAndExit(args: Array[String]): Unit = System.exit(makeItSo(args))
+
+  /** A main method that returns an exit code instead of exiting. */
+  def makeItSo(args: Array[String], packageList: List[String] = DagrCoreMain.getPackageList, includeHidden: Boolean = false): Int = {
+    // Initialize color options
+    TermCode.printColor = DagrCoreMain.optionallyConfigure[Boolean](Configuration.Keys.ColorStatus).getOrElse(true)
+
+    // Load any dagr scripts - assumes the option is "--scripts" only
+    loadScripts(args)
+
+    val startTime = System.currentTimeMillis()
+    val packages = Sopt.find[Pipeline](packageList, includeHidden=includeHidden)
+    val exit      = Sopt.parseCommandAndSubCommand[DagrCoreArgs,Pipeline](name, args, packages) match {
+      case Sopt.Failure(usage) =>
+        System.err.print(usage())
+        1
+      case Sopt.CommandSuccess(cmd) =>
+        unreachable("CommandSuccess should never be returned by parseCommandAndSubCommand.")
+      case Sopt.SubcommandSuccess(dagr, pipeline) =>
+        val name = pipeline.getClass.getSimpleName
+        try {
+          dagr.configure(pipeline, Some(args.mkString))
+          val name = Configuration.commandLineName(this.name)
+          printStartupLines(name, args)
+          val numFailed = dagr.execute(pipeline)
+          if (numFailed!= 0) logger.fatal("Execution failed!")
```

Comment:
I think it's nice when something like this happens to print a larger error message.  E.g. a banner of `###`s followed by the failure, followed by more `###`s.  Just to make it more eye-catching that all did not go as planned.